src/test/java/seedu/address/logic/commands/EditApplicantCommandTest.java

```diff
@@ -0,0 +1,182 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.logic.commands.CommandTestUtil.VALID_APPLICANT_NAME_AMY;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_APPLICANT_NAME_BENEDICT;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_APPLICANT_NAME_CHRIS;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalIndexes.getIndexLastListing;
+import static seedu.address.testutil.TypicalListings.getTypicalListingBook;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.ListingBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.applicant.Applicant;
+import seedu.address.model.listing.Listing;
+import seedu.address.testutil.ApplicantBuilder;
+import seedu.address.testutil.ListingBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for DeleteApplicantCommand.
+ */
+public class EditApplicantCommandTest {
+    private static final Applicant AMY = new ApplicantBuilder().withName(VALID_APPLICANT_NAME_AMY).build();
+    private static final Applicant BENEDICT_1 = new ApplicantBuilder().withName(VALID_APPLICANT_NAME_BENEDICT).build();
+    private static final Applicant BENEDICT_2 = new ApplicantBuilder().withName(VALID_APPLICANT_NAME_BENEDICT).build();
+    private static final Applicant CHRIS = new ApplicantBuilder().withName(VALID_APPLICANT_NAME_CHRIS).build();
+    private static final String INVALID_BENEDICT_1_NAME_WITH_ID = VALID_APPLICANT_NAME_BENEDICT + "#"
+            + (BENEDICT_1.hashCode() - 1);
+    private static final String VALID_BENEDICT_1_NAME_WITH_ID = VALID_APPLICANT_NAME_BENEDICT + "#"
+            + BENEDICT_1.hashCode();
```

Comment:
similar to what i said on top, i dont think users can see the hashcode if the applicant is unique right? so either need to handle the parsing for name only without id or just display id for all applicants, unique or not

Reply:
if dont need the hashcode then dont need to type. like if is just replacing John with Ben is just edit_app 1 id/John a/Ben. sama sama

Reply:
Ok solid if that case is handled then alls good just need explain in UG